src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProject.cs

```diff
@@ -154,8 +155,45 @@ public void It_does_not_contain_apphost_exe(bool multiTarget)
                GetValuesCommand.ValueType.Property);
 
             getValuesCommand.Execute();
-            Path.GetExtension(getValuesCommand.GetValues().Single())
-                .Should().NotBe(".exe", "Repro https://github.com/dotnet/cli/issues/11299");
+            string runCommandPath = getValuesCommand.GetValues().Single();
+            Path.GetExtension(runCommandPath)
+                .Should().Be(".exe");
+            File.Exists(runCommandPath).Should()
+                .BeTrue("run command should be apphost exe for WinExe to debug. But it will not be packed");
+        }
+
+        [PlatformSpecificTheory(Platform.Linux, Platform.Darwin, Platform.FreeBSD)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Given_nonwindows_It_does_not_contain_apphost_and_RunCommand_is_not_apphost(bool multiTarget)
+        {
+            _targetFrameworkOrFrameworks = "netcoreapp3.0";
+
+            var nugetPackage = SetupNuGetPackage(multiTarget);
+            using (var nupkgReader = new PackageArchiveReader(nugetPackage))
+            {
+                IEnumerable<NuGet.Frameworks.NuGetFramework> supportedFrameworks = nupkgReader.GetSupportedFrameworks();
+                supportedFrameworks.Should().NotBeEmpty();
+
+                foreach (NuGet.Frameworks.NuGetFramework framework in supportedFrameworks)
+                {
+                    var allItems = nupkgReader.GetToolItems().SelectMany(i => i.Items).ToList();
+                    allItems.Should().NotContain($"tools/{framework.GetShortFolderName()}/any/consoledemo");
+                }
+            }
+
+            var getValuesCommand = new GetValuesCommand(
+               Log,
+               _testRoot,
+               _targetFrameworkOrFrameworks,
+               "RunCommand",
+               GetValuesCommand.ValueType.Property);
+
+            getValuesCommand.Execute();
+            string runCommandPath = getValuesCommand.GetValues().Single();
+            Path.GetExtension(runCommandPath)
+                .Should().NotBe(".exe");
```

Comment:
Would it ever be .exe on non-Windows?

Reply:
you are right. non-windows and windows can be combined. The only difference is ".exe" vs no extension.

Reply:
could you look at it again? I combined 2 tests